bloptools/experiments/tests/__init__.py

```diff
@@ -1,20 +1,34 @@
 import bluesky.plan_stubs as bps
 import numpy as np
-from ophyd import Signal
+from ophyd import Component as Cpt
+from ophyd import Device, Signal
 
 
 def get_dofs(n=2):
     return [Signal(name=f"x{i+1}", value=0) for i in range(n)]
 
 
+def get_device(name="dofs", n=2):
+    components = {}
+
+    for i in range(n):
+        components[f"{i+1}"] = Cpt(Signal, value=i + 1)
```

Comment:
We need to have the component attribute names starting with `x`, otherwise one will get this error:
```py
In [3]: get_device()
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In [3], line 1
----> 1 get_device()

Cell In [2], line 7, in get_device(name, n)
      4 for i in range(n):
      5     components[f"{i+1}"] = Cpt(Signal, value=i + 1)
----> 7 cls = type("DOF", (Device,), components)
      9 device = cls(name=name)
     11 return [getattr(device, attr) for attr in device.read_attrs]

File ~/anaconda3/envs/sirepo-bluesky/lib/python3.9/site-packages/ophyd/device.py:982, in Device.__init_subclass__(cls, **kwargs)
    980 "This is called automatically in Python for all subclasses of Device"
    981 super().__init_subclass__(**kwargs)
--> 982 cls._initialize_device()

File ~/anaconda3/envs/sirepo-bluesky/lib/python3.9/site-packages/ophyd/device.py:960, in Device._initialize_device(cls)
    957 cls.component_names = tuple(cls._sig_attrs)
    959 # The namedtuple associated with the device
--> 960 cls._device_tuple = namedtuple(
    961     f"{cls.__name__}Tuple",
    962     [comp for comp in cls.component_names[:254] if not comp.startswith("_")],
    963 )
    965 # List the attributes that are Devices (not Signals).
    966 # This list is used by stage/unstage. Only Devices need to be staged.
    967 cls._sub_devices = [
    968     attr for attr, cpt in cls._sig_attrs.items() if cpt.is_device
    969 ]

File ~/anaconda3/envs/sirepo-bluesky/lib/python3.9/collections/__init__.py:390, in namedtuple(typename, field_names, rename, defaults, module)
    388     raise TypeError('Type names and field names must be strings')
    389 if not name.isidentifier():
--> 390     raise ValueError('Type names and field names must be valid '
    391                      f'identifiers: {name!r}')
    392 if _iskeyword(name):
    393     raise ValueError('Type names and field names cannot be a '
    394                      f'keyword: {name!r}')

ValueError: Type names and field names must be valid identifiers: '1'
```
```suggestion
        components[f"x{i+1}"] = Cpt(Signal, value=i + 1)
```